dags/annuaire_entrerpise_integration.py

```diff
@@ -0,0 +1,126 @@
+import gzip
+import io
+from datetime import datetime
+from importlib import import_module
+from pathlib import Path
+
+import pandas as pd
+import requests
+from airflow import DAG
+from airflow.operators.python import PythonOperator
+from airflow.providers.postgres.hooks.postgres import PostgresHook
+from tqdm import tqdm
+
+env = Path(__file__).parent.name
+utils = import_module(f"{env}.utils.utils")
+
+default_args = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "start_date": datetime(2024, 8, 18),
+    "retries": 1,
+    "email_on_failure": False,
+    "email_on_retry": False,
+}
+
+dag = DAG(
+    "annuaire_entreprise_datasets_integration",
+    default_args=default_args,
+    description="Download unités légales and etablissements from data.gouv.fr "
+    "and save it to postgresql",
+    schedule_interval="@weekly",
+)
+
+
+def fetch_and_process_data(url_index, table_name, index_column, schema, **context):
+    pg_hook = PostgresHook(postgres_conn_id=utils.get_db_conn_id(__file__))
+    engine = pg_hook.get_sqlalchemy_engine()
+    conn = pg_hook.get_conn()
+    cursor = conn.cursor()
+
+    cursor.execute(f"TRUNCATE TABLE  {table_name};")
+    conn.commit()
+
+    metadata_url = "https://www.data.gouv.fr/api/1/datasets/donnees-des-entreprises-utilisees-dans-lannuaire-des-entreprises"
+    metadata_response = requests.get(metadata_url)
+    metadata_response.raise_for_status()
+    metadata = metadata_response.json()
+
+    resources = [
+        resource
+        for resource in metadata["resources"]
+        if resource["type"] != "documentation"
+    ]
+    urls = [resource["url"] for resource in resources]
+
+    if not urls:
+        raise ValueError("No data URLs found in the resources.")
+
+    with requests.get(urls[url_index], stream=True) as r:
+        r.raise_for_status()
+        decompressed_file = gzip.GzipFile(fileobj=io.BytesIO(r.content))
```

Comment:
il charge tout le fichier en mémoire ?
Si oui, il faudra tester en preprod si on a assez de mémoire et ne pas subir de OOM

Sinon est-ce qu'il y a moyen d'itérer pour charger le document 10000 lignes par 10000 lignes ?